aws/resource_aws_s3_bucket_test.go

```diff
@@ -1862,6 +1862,66 @@ func TestAccAWSS3Bucket_ReplicationSchemaV2(t *testing.T) {
 	})
 }
 
+func TestAccAWSS3Bucket_SameRegionReplicationSchemaV2(t *testing.T) {
+	rInt := acctest.RandInt()
+	region := testAccGetRegion()
+	partition := testAccGetPartition()
+	resourceName := "aws_s3_bucket.bucket"
+
+	// record the initialized providers so that we can use them to check for the instances in each region
+	var providers []*schema.Provider
```

Comment:
Thanks for adding this test! ❤️ 

Given this test is explicitly for testing within the same region, it seems like we should be able to nix the more complicated `ProviderFactories` setup for a normal test setup and inline both `aws_s3_bucket` resources into the same test configuration without provider aliases, etc. 😄

Reply:
Agreed. I'll simplify.